apps/backend/app/api/routes/datasets.py

```
@@ -821,3 +837,326 @@ async def preview_transformation(
+        dataset = await dataset_service.get_by_id(dataset_id, current_user_id)
+
+        # Check if feature selection results are stored in metadata
+        selected_features = dataset.metadata.get("selected_features", []) if hasattr(dataset, "metadata") else []
+
+        return {
+            "dataset_id": dataset_id,
+            "selected_features": selected_features,
+            "has_selection": len(selected_features) > 0
+        }
+
+    except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)